wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/scm/AbstractGitUserDataFetcher.java

```diff
@@ -27,48 +24,38 @@
  */
 public abstract class AbstractGitUserDataFetcher implements GitUserDataFetcher {
   protected final String oAuthProviderName;
+  private final String oAuthProviderUrl;
   protected final PersonalAccessTokenManager personalAccessTokenManager;
-  protected final OAuthAPI oAuthTokenFetcher;
 
   public AbstractGitUserDataFetcher(
       String oAuthProviderName,
-      PersonalAccessTokenManager personalAccessTokenManager,
-      OAuthAPI oAuthTokenFetcher) {
+      String oAuthProviderUrl,
+      PersonalAccessTokenManager personalAccessTokenManager) {
     this.oAuthProviderName = oAuthProviderName;
+    this.oAuthProviderUrl = oAuthProviderUrl;
     this.personalAccessTokenManager = personalAccessTokenManager;
-    this.oAuthTokenFetcher = oAuthTokenFetcher;
   }
 
   public GitUserData fetchGitUserData()
       throws ScmUnauthorizedException, ScmCommunicationException,
           ScmConfigurationPersistenceException, ScmItemNotFoundException, ScmBadRequestException {
     Subject cheSubject = EnvironmentContext.getCurrent().getSubject();
-    try {
-      OAuthToken oAuthToken = oAuthTokenFetcher.getToken(oAuthProviderName);
```

Comment:
Do we ignore the case when token exists in memory storage but not in secret?

Reply:
Yes, this is removed to avoid circular dependencies: we can not use the `oauth` module in the `factory` module any more since we are using `PersonalAccessTokenManager` in the `EmbeddedOAuthAPI`. I don't see a problem here because this fetcher works only in case no gitconfig configmap is setup. So even if user deletes all PATs the configmap would be present and this request would be skiped.